python/openmldb_tool/diagnostic_tool/server_checker.py

```diff
@@ -86,75 +88,91 @@ def is_exist(self, data, name):
 
     def get_job_status(self, job_id):
         try:
-            result = self.cursor.execute(
-                'SHOW JOB {};'.format(job_id)).fetchall()
+            result = self.cursor.execute("SHOW JOB {};".format(job_id)).fetchall()
             return result[0][2]
         except Exception as e:
-            log.warn(e)
+            log.warning(e)
             return None
 
-    def check_run_job(self) -> bool:
-        if 'taskmanager' not in self.conf_dict:
-            log.info('no taskmanager installed. skip job test')
+    def check_offline_select(self) -> bool:
+        if "taskmanager" not in self.conf_dict:
+            log.info("no taskmanager installed. skip job test")
             return True
-        self.cursor.execute('SET @@execute_mode=\'offline\';')
+        self.cursor.execute("SET @@execute_mode='offline';")
+        # make sure it's an async job, to get the job id, not the select result
+        self.cursor.execute("SET @@sync_job=false;")
         result = self.cursor.execute(
-            'SELECT * FROM {};'.format(self.table_name)).fetchall()
-        if len(result) < 1:
-            log.warn('run job failed. no job info returned')
+            "SELECT * FROM {};".format(self.table_name)
+        ).fetchall()
+        if len(result) != 1:
+            log.warning(f"result is invalid, expect a job info: {result}")
             return False
-        job_id = result[0][0].split('\n')[3].strip().split(' ')[0]
-        time.sleep(2)
-        while True:
+        # see system table JOB_INFO schema, job id idx is 0
+        job_id, try_time = result[0][0], 0
```

Comment:
update the required version in setup.py ? https://github.com/4paradigm/OpenMLDB/blob/main/python/openmldb_tool/setup.py#L31

Reply:
done